extension/src/popup/components/manageNetwork/NetworkForm/index.tsx

```diff
@@ -23,60 +38,202 @@ interface FormValues {
   const dispatch: AppDispatch = useDispatch();
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const networksList = useSelector(settingsNetworksListSelector);
+  const settingsError = useSelector(settingsErrorSelector);
+  const [isNetworkInUse, setIsNetworkInUse] = useState(false);
+  const [isConfirmingRemoval, setIsConfirmingRemoval] = useState(false);
+  const history = useHistory();
+  const { search } = useLocation();
+
+  const networkIndex = new URLSearchParams(search).get("networkIndex");
+  const networkDetailsToEdit = networksList[Number(networkIndex)];
+  const isCurrentNetworkActive = isActiveNetwork(
+    networkDetailsToEdit,
+    networkDetails,
+  );
+  const isMainnetOrTestnet =
+    isMainnet(networkDetailsToEdit) || isTestnet(networkDetailsToEdit);
+
+  const initialValues: FormValues = isEditing
+    ? { ...networkDetailsToEdit, isSwitchSelected: false }
+    : {
+        networkName: "",
+        networkPassphrase: "",
+        networkUrl: "",
+        isSwitchSelected: false,
+      };
 
-  const initialValues: FormValues = {
-    networkName: "",
-    networkPassphrase: "",
-    networkUrl: "",
-    isSwitchSelected: false,
+  const NetworkFormSchema = YupObject().shape({
+    networkName: YupString().required(),
+    networkPassphrase: YupString().required(),
+    networkUrl: YupString().required(),
+  });
+
+  const handleRemoveNetwork = async () => {
+    const res = await dispatch(
+      removeCustomNetwork({
+        networkName: networkDetailsToEdit.networkName,
+      }),
+    );
+
+    if (removeCustomNetwork.fulfilled.match(res)) {
+      navigateTo(ROUTES.account);
+    }
   };
 
-  const handleSubmit = async (values: FormValues) => {
-    if (isEditing) {
-      const res = await dispatch(changeNetwork(values));
-      if (changeNetwork.fulfilled.match(res)) {
-        navigateTo(ROUTES.account);
-      }
+  const handleEditNetwork = async (values: FormValues) => {
+    if (isCurrentNetworkActive) {
+      setIsNetworkInUse(true);
     } else {
       const res = await dispatch(
-        addCustomNetwork({
-          customNetwork: {
-            network: CUSTOM_NETWORK,
-            ...values,
-          },
+        editCustomNetwork({
+          networkDetails: { ...values, network: CUSTOM_NETWORK },
+          networkIndex: Number(networkIndex),
         }),
       );
-      if (addCustomNetwork.fulfilled.match(res)) {
+      if (editCustomNetwork.fulfilled.match(res)) {
         navigateTo(ROUTES.account);
       }
     }
   };
 
+  const handleAddNetwork = async (values: FormValues) => {
+    const addCustomNetworkRes = await dispatch(
+      addCustomNetwork({
+        networkDetails: {
+          ...values,
+          network: CUSTOM_NETWORK,
+        },
+      }),
+    );
+
+    const addCustomNetworkFulfilled = addCustomNetwork.fulfilled.match(
+      addCustomNetworkRes,
+    );
+    let changeNetworkFulfilled = true;
+
+    if (values.isSwitchSelected) {
```

Comment:
if the user wants to automatically switch to the new network after adding, we just call `changeNetwork` next